src/core/net/dnssd_server.cpp

```diff
@@ -886,7 +890,11 @@ Server::UpstreamQueryTransaction *Server::AllocateUpstreamQueryTransaction(const
         }
     }
 
-    VerifyOrExit(newTxn != nullptr);
+    if (newTxn == nullptr)
+    {
+        mCounters.mUpstreamDnsCounters.mFailures++;
+        ExitNow();
+    }
```

Comment:
```suggestion
    VerifyOrExit (newTxn != nullptr, mCounters.mUpstreamDnsCounters.mFailures++);
```
Could be simplified like this.